src/main/scala/io/iohk/ethereum/domain/branch/BestBlockchainBranch.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.ethereum.domain.branch
+import akka.util.ByteString
+
+import io.iohk.ethereum.db.storage.BlockNumberMappingStorage
+import io.iohk.ethereum.domain.Block
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.domain.BlockchainReader
+
+/** A Branch instance which only works for the best canonical branch. As this branch
+  * currently has specific indexes (particularly regarding accessing blocks by number),
+  * it will uses thoses to provide better performance.
```

Comment:
```suggestion
  * it will use thoses to provide better performance.
```
and what is `thoses`?

Reply:
Oops, Its from an old comment I removed.

Reply:
I referred to the index, but the comment was initially longer. I'll rephrase that